scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -319,6 +319,11 @@ class FossId(
             val filteredPackages = packages
                 .filter { packageNamespaceFilter.isEmpty() || it.id.namespace == packageNamespaceFilter }
                 .filter { packageAuthorsFilter.isEmpty() || packageAuthorsFilter in it.authors }
+                .onEach {
+                    if (it.vcs.path.isNotEmpty()) {
+                        log.warn { "Ignoring package with url ${it.vcs.url} with non-null path ${it.vcs.path}" }
+                    }
+                }.filter { it.vcsProcessed.path.isEmpty() }
```

Comment:
Is this tested anywhere?
To me it looks like you first check if a path is set in the vcs.path, stating that it will be ignored, and then filter out all the path of vcs processed that are NOT set. 
This seems counter-intuitive to your commit message.